packages/blockly/src/sims/predator-prey-code.js

```diff
@@ -133,6 +133,15 @@ function setup() {
   create_wolves(10);
 }
 
+// Agents added outside of the sim tick function (like onClick) do not get added to the renderer,
+// so instead of adding agents directly, we create a queue and add them in beforeTick.
+let actorsToAdd = [];
+
+sim.beforeTick = () => {
+  actorsToAdd.forEach(a => a.addTo(sim));
+  actorsToAdd = [];
+}
```

Comment:
This is more of a question than a suggestion. I'm pretty sure the current code is perfectly fine as is. But is there a reason you didn't do `const actorsToAdd = [];` on line 138, and then something like `actorsToAdd.length = 0;` to empty the array on line 142?

I know it's not _necessary_, but I've had it beaten into my brain that using `const` when you can is best. Just wondering what the reasoning is here. Maybe the current code is just more self-explanatory? (`actorsToAdd = []` is definitely clearer than `actorsToAdd.length = 0`.)